src/intrinsics/mod.rs

```diff
@@ -202,14 +212,33 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                     _ => bug!(),
                 };
                 let res = f.round_to_integral(mode).value;
+                // Apply a relative error with a magnitude on the order of 2^-12 to simulate
+                // non-deterministic behaviour of floats
+                let res = apply_random_float_error(this, res, -12);
+                let res = this.adjust_nan(res, &[f]);
+                this.write_scalar(res, dest)?;
+            }
+
+            "sqrtf32" => {
+                let [f] = check_arg_count(args)?;
+                let f = this.read_scalar(f)?.to_f32()?;
+                // no host floats for sqrt
```

Comment:
```suggestion
                // Sqrt is specified to be fully precise.
```
(same below for sqrtf64)